src/store/modules/search.store.js

```diff
@@ -51,7 +55,27 @@ const actions = {
                 }
             }
 
-            if (coordinates) {
+            if (extractedCoordinate) {
+                let coordinates = [...extractedCoordinate.coordinate]
+                if (extractedCoordinate.coordinateSystem !== currentProjection) {
+                    // special case for LV03 input, we can't use proj4 to transform them into
+                    // LV95 or others, as the deformation between LV03 and the others is not constant.
+                    // So we pass through a LV95 reframe (done by a backend service that knows all deformations between the two)
+                    // and then go to the wanted coordinate system
+                    if (extractedCoordinate.coordinateSystem === LV03) {
+                        coordinates = await reframe({
+                            inputProjection: LV03,
+                            inputCoordinates: coordinates,
+                        })
+                        extractedCoordinate.coordinateSystem = LV95
+                    }
+
+                    coordinates = proj4(
+                        extractedCoordinate.coordinateSystem.epsg,
+                        currentProjection.epsg,
+                        coordinates
+                    ).map(currentProjection.roundCoordinateValue)
```

Comment:
Not sure that overwriting the `extractedCoordinate.coordinateSystem = LV95` is a good idea as well as passing the reframe lv03 to lv95 to proj4, make the code a bit hard to understand and error prone. Would be better to have an else statement for the proj4 projection and not overwrite the extractedCoordinate.